webapp/tests/karma/ts/reducers/tasks.spec.ts

```diff
@@ -242,15 +242,215 @@ describe('Tasks reducer', () => {
         },
         tasksList: [
           { _id: 'task9', dueDate: -100, state: 'Ready', field: 9 },
+          { _id: 'task3', dueDate: 0, state: 'Ready', field: 3 },
           { _id: 'task7', dueDate: 125, state: 'Ready', field: 7 },
           { _id: 'task6', dueDate: 250, state: 'Ready', field: 6 },
           { _id: 'task4', dueDate: 500, state: 'Ready', field: 4 },
           { _id: 'task5', dueDate: 500, state: 'Ready', field: 5 },
           { _id: 'task8', dueDate: 899, state: 'Ready', field: 8 },
           { _id: 'task1', dueDate: false, state: 'Ready', field: 1 },
-          { _id: 'task2', dueDate: undefined, state: 'Ready', field: 2 },
-          { _id: 'task3', dueDate: 0, state: 'Ready', field: 3 },
-        ]
+          { _id: 'task2', dueDate: null, state: 'Ready', field: 2 },
+        ],
+      });
+    });
+
+    it('should sort provided tasks by priority and due date', () => {
+      const tasks = [
+        {
+          _id: 'task1',
+          dueDate: '2025-05-30',
+          priority: 'invalid',
+          state: 'Ready',
+          field: 1,
```

Comment:
Can you please add a task that has both invalid due date _and_ invalid priority? Also please don't add it as last in the initial list, it would be good to see it actually pushed down to the end.